main.py

```diff
@@ -164,6 +150,51 @@ def draw(self):
         img = g_font.render("Moves: " + str(self.counter), True, BLACK)
         g_screen.blit(img, (20, 20))
 
+    def move(self, ball):
+        """
+        Function Game.move(self, ball):
+        Move function, that handles handles the action when the player clicks on 
+        a ball depending on the current ball state. The three cases are described
+        below.
+
+        param[in]   ball    clicked ball (Ball)
+        """
+
+        # Case(1): No ball is selected atm and the observed ball is not white.
+        #   -> Select it and set depending variables accordingly.
+        if not self.isBallSelected and ball.colour != 0:
+            ball.select()
+            checkBalls = []
+            checkBalls.append(ball.id)
+            while len(checkBalls) > 0:
+                for selectableBall in self.balls[checkBalls[0]].neighbours:
+                    if (self.balls[selectableBall].colour == 0 and 
+                        self.balls[selectableBall].selected == BALL_NOMARK):
+                        self.balls[selectableBall].selected = BALL_MARK_S
+                        checkBalls.append(selectableBall)
+                checkBalls.pop(0)
+
+            self.selectedBall = ball
+            self.isBallSelected = True
+
+        # Case(2): A ball is selected atm and the observed ball has a big rim.
+        #   -> Unselect all balls and set depending variables accordingly.
+        elif self.isBallSelected and ball.selected == BALL_MARK_L:
+            for ball in self.balls:
+                ball.select(False)
+            self.isBallSelected = False
+
+        # Case(3): A ball is selected atm and the observed ball has a small rim.
+        #   -> Swap the colours of the selected and the observed ball, unselect
+        #      all balls and set depending variables accordingly.
+        elif self.isBallSelected and ball.selected == BALL_MARK_S:
+            ball.colour = self.selectedBall.colour
+            self.selectedBall.colour = 0
```

Comment:
Does using the macro makes sense here?
```suggestion
            self.selectedBall.colour = N
```

Reply:
adapted